akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -522,8 +522,9 @@ final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out) extends GraphSta
           pull(in)
         } catch {
           case NonFatal(ex) ⇒ decider(ex) match {
-            case Supervision.Stop ⇒ failStage(ex)
-            case _ ⇒
+            case Supervision.Stop   ⇒ failStage(ex)
```

Comment:
Doesn't  it make sense to extract pull into finally block? pull can potentially generate exception itself and catching it is not what we want to do here.